homeassistant/components/isy994/climate.py

```diff
@@ -0,0 +1,360 @@
+            self._current_humidity = current_humidity['value']
+
+        target_temp_high = next((i for i in self._node.aux_properties
+                                 if i['id'] == ISY_TARGET_TEMP_HIGH), False)
+        if target_temp_high:
+            self._target_temp_high = self._fix_temp(target_temp_high['value'])
+
+        target_temp_low = next((i for i in self._node.aux_properties
+                                if i['id'] == ISY_TARGET_TEMP_LOW), False)
+        if target_temp_low:
+            self._target_temp_low = self._fix_temp(target_temp_low['value'])
+
+        current_operation = next((i for i in self._node.aux_properties
+                                  if i['id'] == ISY_OPERATION_MODE), False)
+        if current_operation:
+            self._current_isy_operation = VALUE_TO_ISY_MODE.get(
+                current_operation['value'], None)
+            self._current_operation, self._current_hold_mode = \
+                VALUE_TO_HASS_MODE.get(current_operation['value'], None)
+
+        await super().async_added_to_hass()
+        self.async_schedule_update_ha_state()
+        self._node.controlEvents.subscribe(self._node_control_handler)
+
+    def _node_control_handler(self, event: object) -> None:
+        """Handle control events coming from the primary node.
+
+        The ISY does not report some properties on the root of the node,
+            they only show up in the event log:
+
+        ISY_FAN_MODE, ISY_OPERATION_STATE, ISY_UOM will be set the first
+            time the event is fired by the ISY for those controls.
+
+        Current Temperature is updated by PyISY in node.status and we don't
+            need to listen for it here.
+        """
+        if event.event == ISY_FAN_MODE:
+            self._fan_mode = FAN_VALUE_TO_MODE.get(int(event.nval),
+                                                   None)
+        elif event.event == ISY_OPERATION_STATE:
+            self._current_status = VALUE_TO_STATE.get(int(event.nval),
+                                                      None)
+        elif event.event == ISY_OPERATION_MODE:
+            self._current_isy_operation = \
+                VALUE_TO_ISY_MODE.get(int(event.nval), None)
+            self._current_operation, self._current_hold_mode = \
+                VALUE_TO_HASS_MODE.get(int(event.nval), None)
+        elif event.event == ISY_UOM:
+            if int(event.nval) == 1:
+                self._temp_unit = TEMP_CELSIUS
+            elif int(event.nval) == 2:
+                self._temp_unit = TEMP_FAHRENHEIT
+        elif event.event == ISY_CURRENT_HUMIDITY:
+            self._current_humidity = int(event.nval)
+        elif event.event == ISY_TARGET_TEMP_HIGH:
+            self._target_temp_high = self._fix_temp(int(event.nval))
+        elif event.event == ISY_TARGET_TEMP_LOW:
+            self._target_temp_low = self._fix_temp(int(event.nval))
+        self.schedule_update_ha_state()
+
+    @property
+    def state(self):
```

Comment:
Don't overwrite this. It's the base climate entity's responsibility. Implement `current_operation` and/or `is_on` as needed.

Reply:
For this I'd like to see some resolution to home-assistant/architecture#22 before final acceptance.  In my opinion, current state ≠ current operation.  The base climate entity currently uses `current_operation` as it's state, but that creates mis-matched information:  I can have my current_operation mode set to `HEAT`, but the furnace might not be on at the moment because it is within temperature range--if I am reporting `state` as `current_operation` I have no way of knowing if the unit is actually running or not.

Reply:
There's no timeline for arch issue 22. I would suggest you follow my change request so we can get this PR merged. When arch issue 22 is resolved we can update the integration.